tests/foreman/api/test_discoveredhost.py

```diff
@@ -547,42 +339,92 @@ def test_positive_reboot_all_pxe_hosts(
-                        except TimedOutError:
-                            # raise assertion error
-                            raise AssertionError(
-                                f'Timed out waiting for {pattern[1]} from ' f'{pxe_host[0].mac}'
-                            )
+
+
+class TestFakeDiscoveryTests:
+    """Tests that uses fake discovered host."""
+
+    def _create_discovered_host(self, name=None, ipaddress=None, macaddress=None):
+        """Creates discovered host by uploading few fake facts.
+
+        :param str name: Name of discovered host. If ``None`` then a random
+            value will be generated.
+        :param str ipaddress: A valid ip address.
+            If ``None`` then then a random value will be generated.
+        :param str macaddress: A valid mac address.
+            If ``None`` then then a random value will be generated.
+        :return: A ``dict`` of ``DiscoveredHost`` facts.
+        """
+        if name is None:
+            name = gen_string('alpha')
+        if ipaddress is None:
+            ipaddress = gen_ipaddr()
+        if macaddress is None:
+            macaddress = gen_mac(multicast=False)
+        return entities.DiscoveredHost().facts(
+            json={
+                'facts': {
+                    'name': name,
+                    'discovery_bootip': ipaddress,
+                    'discovery_bootif': macaddress,
+                    'interfaces': 'eth0',
+                    'ipaddress': ipaddress,
+                    'macaddress': macaddress,
+                    'macaddress_eth0': macaddress,
+                    'ipaddress_eth0': ipaddress,
+                }
+            }
+        )
+
+    @pytest.mark.tier2
+    def test_positive_upload_facts(self):
+        """Upload fake facts to create a discovered host
+
+        :id: c1f40204-bbb0-46d0-9b60-e42f00ad1649
+
+        :BZ: 1349364, 1392919
+
+        :Steps:
+
+            1. POST /api/v2/discovered_hosts/facts
+            2. Read the created discovered host
+
+        :expectedresults: Host should be created successfully
+
+        :CaseImportance: High
+
+        :CaseLevel: Integration
+
+        :BZ: 1731112
+        """
+        name = gen_choice(list(valid_data_list().values()))
+        result = self._create_discovered_host(name)
+        discovered_host = entities.DiscoveredHost(id=result['id']).read_json()
+        host_name = 'mac{}'.format(discovered_host['mac'].replace(':', ''))
+        assert discovered_host['name'] == host_name
+
+    @pytest.mark.tier3
+    def test_positive_delete_pxe_host(self):
+        """Delete a pxe-based discovered hosts
+
+        :id: 2ab2ad88-4470-4d4c-8e0b-5892ad8d675e
+
+        :Setup: Provisioning should be configured and a host should be
+            discovered
+
+        :Steps: DELETE /api/v2/discovered_hosts/:id
+
+        :expectedresults: Discovered Host should be deleted successfully
+
+        :CaseAutomation: Automated
+
+        :CaseImportance: High
```

Comment:
You can move this docstring at class level as Its common for both of ur tests.

Non-blocking comment!